workshop/content/intermediate/operations/understanding-changesets/_index.md

```diff
@@ -0,0 +1,259 @@
+
+You will now modify the value for a property, `BucketName`, that requires a [replacement](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html#update-replacement) on stack updates. You will then create a change set to preview your changes, and inspect the output of the change set operation.
+
+Let’s get started!
+
+1. In the CloudFormation console, select the `changesets-workshop` stack, and from **Stack actions**, choose **Create change set for current stack**.
+2. From **Prepare template**, choose **Use current template** and choose **Next**.
+3. Change the value for `BucketName` parameter by specifying a new unique bucket [name](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html), and follow the rest of the process as before to finish creating the change set.
+
+Here’s what the **JSON changes** for this change set should look like:
+
+
+```json
+[
+  {
+    "resourceChange": {
+      "logicalResourceId": "MyS3Bucket",
+      "action": "Modify",
+      "physicalResourceId": "understanding-changesets-123",
+      "resourceType": "AWS::S3::Bucket",
+      "replacement": "True",
+      "moduleInfo": null,
+      "details": [
+        {
+          "target": {
+            "name": "BucketName",
+            "requiresRecreation": "Always",
+            "attribute": "Properties"
+          },
+          "causingEntity": null,
+          "evaluation": "Dynamic",
+          "changeSource": "DirectModification"
+        },
+        {
+          "target": {
+            "name": "BucketName",
+            "requiresRecreation": "Always",
+            "attribute": "Properties"
+          },
+          "causingEntity": "BucketName",
+          "evaluation": "Static",
+          "changeSource": "ParameterReference"
+        }
+      ],
+      "changeSetId": null,
+      "scope": [
+        "Properties"
+      ]
+    },
+    "hookInvocationCount": null,
+    "type": "Resource"
+  }
+]
+```
+
+You can see there are two key differences from the previous example. First, the value for the property `replacement` under `resourceChange` structure is `True` and second, you see two evaluations `Static` and `Dynamic` under the `details` structure. Let's talk about these in more detail.
+
+The value for `replacement` is `True` because you updated the `BucketName` [property](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-s3-bucket.html#cfn-s3-bucket-bucketname) that requires a replacement. CloudFormation will create a new resource (a new bucket in this case), and then delete the old one. If there are multiple changes you make on a given resource, and each change has a different value for the `requiresRecreation` field, CloudFormation updates the resource when a recreation is required. In other words, if only one of the many changes requires a replacement, CloudFormation replaces the resources, and therefore sets the `replacement` field to `True`.
```

Comment:
```suggestion
The value for `replacement` is `True` because you updated the `BucketName` [property](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-s3-bucket.html#cfn-s3-bucket-bucketname) that requires a replacement. CloudFormation will create a new resource (a new bucket in this case), and then delete the old one. If there are multiple changes you make on a given resource, and each change has a different value for the `requiresRecreation` field, CloudFormation replaces the resource when a recreation is required. In other words, if only one of the many changes requires a replacement, CloudFormation replaces the resources, and therefore sets the `replacement` field to `True`.
```